python/lsst/daf/persistence/butler.py

```diff
@@ -433,8 +451,18 @@ def _read(self, pythonType, location):
         return results
 
     def __reduce__(self):
-        return (_unreduce, (self.mapper,))
+        return (_unreduce, (self.mapper, self.datasetTypeAliasDict))
+
+    def __resolveDatasetTypeAlias(self, type):
+        resolvedType = type
```

Comment:
I think this whole method could be replaced by `return self.datasetTypeAliasDict.get(datasetType, datasetType)`.  But it doesn't quite do enough.  First of all, the specification is to replace "`@alias`" (not just "`alias`") with "`datasetType`".  Second, the specification is also to replace "`@alias`" when it occurs as a *substring* of the original type: "`@coadd`" -> "`deepStack`" also implies "`@coadd_src`" -> "`deepStack_src`".